ui/packages/security-idp-server/src/main/webapp/js/view/Login.view.js

```diff
@@ -64,9 +64,17 @@ define([
       $.ajax({
         type: 'GET',
         url: './login/sso',
-        async: false,
+        async: true,
         data: this.model.toJSON(),
         beforeSend: function(xhr) {
+          $('#loginerrorup').hide()
+          $('#loginspinner').show()
+          $('#username').attr('readonly', true)
+          $('#password').attr('readonly', true)
+          $('.btn-upsignin').attr('disabled', true)
+          $('.btn-upclear').attr('disabled', true)
+          $('.btn-pkisignin').attr('disabled', true)
+          $('.btn-guestsignin').attr('disabled', true)
```

Comment:
✏️ May improve readability if these element changes were extracted into their own method with a descriptive name. For example:
```
    showSpinnerAndDisableInput: function() {
          $('#loginerrorup').hide()
          $('#loginspinner').show()
          $('#username').attr('readonly', true)
          $('#password').attr('readonly', true)
          $('.btn-upsignin').attr('disabled', true)
          $('.btn-upclear').attr('disabled', true)
          $('.btn-pkisignin').attr('disabled', true)
          $('.btn-guestsignin').attr('disabled', true)
    }
```